### PR TITLE
hplip: Fix missing library

### PR DESCRIPTION
**Pull request by bkchr**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bkchr** [comment]
CC @flokli

**Comment by flokli** [comment]
@bkchr Thanks for the PR! Can you elaborate a bit in how this fails? I was actually able to build this, so I'm curious…

**Comment by vcunat** [comment]
Like this?  https://hydra.nixos.org/build/122332330

**Comment by flokli** [comment]
@vcunat no, it failed on master yesterday, and with https://github.com/NixOS/nixpkgs/pull/89120 merged, didn't. It currently builds on master.

**Comment by bkchr** [comment]
Sorry, I should have made it clear. It builds, but it fails at running hp-setup for example. Other commands probably fail as well.

**Comment by flokli** [comment]
Hmm, even with that patch, running the application (from another nixpkgs checkout) fails, as the QT stuff isn't wrapped:

```
HP Linux Imaging and Printing System (ver. 3.20.5)
Printer/Fax Setup Utility ver. 9.0

Copyright (c) 2001-18 HP Development Company, LP
This software comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to distribute it
under certain conditions. See COPYING file for more details.

qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

[1]    4543 abort (core dumped)  result/bin/hp-setup
```

**Comment by flokli** [comment]
I updated the commit message to include some more detail, and pushed a second commit doing the qt wrapping for the hplip binaries. PTAL.

**Comment by bkchr** [comment]
Hmm. @flokli you probably not use KDE? That is maybe the reason why I did not get the problems with QT.

**Comment by flokli** [comment]
Nah, the problem can also happen in other cases, too: https://github.com/NixOS/nixpkgs/issues/65399

File: pkgs/misc/drivers/hplip/default.nix [hunk 1]
```diff
@@ -219,6 +222,10 @@ python3Packages.buildPythonApplication {
       --replace {,${utillinux}/bin/}logger \
       --replace {/usr,$out}/bin
     remove-references-to -t ${stdenv.cc.cc} $(readlink -f $out/lib/*.so)
+  '' + stdenv.lib.optionalString withQt5 ''
+    for f in $out/bin/hp-*;do
+      wrapQtApp $f
```
Comment (doronbehar) [inline pkgs/misc/drivers/hplip/default.nix:227]: FYI, I'm pretty sure the binaries are wrapped twice now. It should have been:

```
  preFixup = ''
    makeWrapperArgs+=("''${qtWrapperArgs[@]}")
  ''
```

But it's not that bad...

Comment (flokli) [reply pkgs/misc/drivers/hplip/default.nix:227]: Yeah, if someone wants to clean this up, I'd happily review a PR, but don't feel it a pressing enough issue to do it myself :-)

**Pull request merged**